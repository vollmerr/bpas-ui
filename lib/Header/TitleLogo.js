var _templateObject = _taggedTemplateLiteral(['\n  height: ', 'px;\n  padding: ', 'px;\n'], ['\n  height: ', 'px;\n  padding: ', 'px;\n']);

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import logo from './_caGovLogo.svg';
import theme from '../theme';

var Image = styled.img(_templateObject, theme.size.titleBar, theme.size.xs);

/** Renders the title bars logo  */
function TitleLogo(_ref) {
  var onClick = _ref.onClick;

  return React.createElement(
    'a',
    { href: 'http://www.ca.gov', onClick: onClick },
    React.createElement(Image, { src: logo, alt: 'CA Gov Logo' })
  );
}

TitleLogo.propTypes = {
  onClick: PropTypes.func
};

export default TitleLogo;