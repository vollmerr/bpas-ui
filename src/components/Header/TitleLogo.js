import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import logo from './_caGovLogo.svg';
import theme from '../theme';

const Image = styled.img`
  height: ${theme.size.titleBar}px;
  padding: ${theme.size.xs}px;
`;

/** Renders the title bars logo  */
function TitleLogo({
  onClick,
}) {
  return (
    <a href={'http://www.ca.gov'} onClick={onClick}>
      <Image src={logo} alt={'CA Gov Logo'} />
    </a>
  );
}

TitleLogo.propTypes = {
  onClick: PropTypes.func,
};

export default TitleLogo;
