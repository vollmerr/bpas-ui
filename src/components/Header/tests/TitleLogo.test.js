import React from 'react';
import renderer from 'react-test-renderer';
import TitleLogo from '../TitleLogo';

describe('<TitleLogo />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <TitleLogo />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  xit('passes the correct props down', () => {});
});
