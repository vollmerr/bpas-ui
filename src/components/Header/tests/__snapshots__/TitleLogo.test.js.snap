// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleLogo /> renders correctly 1`] = `
<a
  href="http://www.ca.gov"
  onClick={undefined}
>
  <img
    alt="CA Gov Logo"
    className="sc-bdVaJa jbboGG"
    src="_caGovLogo.svg"
  />
</a>
`;
